pyo3-macros-backend/src/pyproto.rs

```diff
@@ -64,33 +64,22 @@ fn impl_proto_impl(
             if let Some(m) = proto.get_method(&met.sig.ident) {
                 let fn_spec = FnSpec::parse(&mut met.sig, &mut met.attrs, false)?;
 
+                let flags = if m.can_coexist {
+                    // We need METH_COEXIST here to prevent __add__  from overriding __radd__
+                    Some(quote!(pyo3::ffi::METH_COEXIST))
+                } else {
+                    None
+                };
+
                 let method = if let FnType::Fn(self_ty) = &fn_spec.tp {
-                    pymethod::impl_proto_wrap(ty, &fn_spec, &self_ty)?
+                    pymethod::impl_py_method_def(ty, &fn_spec, &self_ty, flags)?
```

Comment:
@kngwyu making this change appears to have partially fixed #1465

The `__bytes__` text signature is still not generated automatically, but the `#[text_signature]` attribute does seem to work now.

Reply:
Great, thanks!